core/types/celo_transaction_signing.go

```diff
@@ -85,7 +85,7 @@ func (s cel2Signer) Hash(tx *Transaction) common.Hash {
 				tx.Value(),
 				tx.Data(),
 				tx.AccessList(),
-				tx.FeeCurrency(),
+				tx.inner.(*CeloDynamicFeeTx).FeeCurrency,
```

Comment:
We'll do this in a lot more places (e.g. debit, credit, intrinsic gas calc), right? Maybe we can find some nice syntactic sugar for it.

Reply:
Nothing comes to mind. You have this type assertion and there is the type switch, but I think using a type switch would be more verbose.

Reply:
How would this look in a case (e.g. the state transition) where we only have the transaction and multiple transaction types. Wouldn't we have to do something like the following? Can we move that into a function on the `Transaction` type?
```go
tx64, ok64 := tx.inner.(*Celo64Tx)
tx66, ok66 := tx.inner.(*Celo66Tx)

if ok64 {
  foo(tx64.FeeCurrency)
} else if ok66 {
  bar(tx66.FeeCurrency)
} else {
  fob()
}
```

Reply:
Ahh, ok yep that sounds like a nice improvement. I'll make that change and see how it looks

Reply:
Done, looking nice!